src/main/java/com/faforever/api/user/UserController.java

```diff
@@ -26,8 +28,34 @@ public void register(@RequestParam("username") String username,
   }
 
   @ApiOperation("Activates a previously registered account.")
-  @RequestMapping(path = "/activate", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  @RequestMapping(path = "/activate", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
   public void activate(@RequestParam("token") String token) {
     userService.activate(token);
   }
+
+  @PreAuthorize("#oauth2.hasScope('change_password') and hasRole('ROLE_USER')")
+  @ApiOperation("Changes the password of a previously registered account.")
+  @RequestMapping(path = "/changePassword", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  public void changePassword(@RequestParam("newPassword") String newPassword, Authentication authentication) {
+    userService.changePassword(newPassword, userService.getUser(authentication));
+  }
+
+  @PreAuthorize("#oauth2.hasScope('change_login') and hasRole('ROLE_USER')")
+  @ApiOperation("Changes the login of a previously registered account.")
+  @RequestMapping(path = "/changeLogin", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  public void changeLogin(@RequestParam("newLogin") String newLogin, Authentication authentication) {
+    userService.changeLogin(newLogin, userService.getUser(authentication));
+  }
+
+  @ApiOperation("Sends a password reset to the email linked by this account.")
+  @RequestMapping(path = "/resetPassword", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  public void resetPassword(@RequestParam("email") String email) {
```

Comment:
The old API requires correct `login` and `email` https://github.com/FAForever/api/blob/develop/api/users_route.py#L141

Reply:
According to the many feedback we received in the last month it should be either `login` or `email`. It seems to be incredibly difficult for people to remember their username and/or email address...